src/storage/raw/store.rs

```diff
@@ -0,0 +1,194 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use super::ttl::TTLSnapshot;
+
+use crate::storage::kv::{Result, ScanMode, Snapshot};
+use crate::storage::Statistics;
+
+use engine_traits::{CfName, IterOptions, DATA_KEY_PREFIX_LEN};
+use txn_types::{Key, KvPair};
+
+pub enum RawStore<S: Snapshot> {
```

Comment:
How about change this into a dynamic trait?

Reply:
Introduce extra overhead